src/coord/src/sink_connector.rs

```diff
@@ -74,16 +71,62 @@ async fn build_kafka(
     // TODO(benesch): do we need to delete the Kafka topic if publishing the
     // schema fails?
     // TODO(sploiselle): support SSL auth'ed sinks
-    let schema_id = ccsr::ClientConfig::new(builder.schema_registry_url)
-        .build()
-        .publish_schema(&format!("{}-value", topic), &builder.value_schema)
+    let schema_id = ccsr
+        .publish_schema(&format!("{}-value", topic), schema)
         .await
         .with_context(|e| format!("unable to publish schema to registry in kafka sink: {}", e))?;
 
+    Ok(schema_id)
+}
+
+async fn build_kafka(
+    builder: KafkaSinkConnectorBuilder,
+    with_snapshot: bool,
+    frontier: Antichain<Timestamp>,
+    id: GlobalId,
+) -> Result<SinkConnector, failure::Error> {
+    let topic = format!("{}-{}-{}", builder.topic_prefix, id, builder.topic_suffix);
+
+    // Create Kafka topic with single partition.
+    let mut config = ClientConfig::new();
+    config.set("bootstrap.servers", &builder.broker_url.to_string());
+    let ccsr = ccsr::ClientConfig::new(builder.schema_registry_url).build();
+
+    let schema_id = register_kafka_topic(
+        &mut config,
+        &topic,
+        builder.replication_factor as i32,
+        &ccsr,
+        &builder.value_schema,
+    )
+    .await
+    .with_context(|e| format!("error registering kafka topic for sink: {}", e))?;
+
+    let consistency = if let Some(consistency_value_schema) = builder.consistency_value_schema {
+        let consistency_topic = format!("{}-consistency", topic);
```

Comment:
maybe we want a better name here?

Reply:
are we using `{}-consistency` for sources, too? should it mention sink?

Reply:
The user specifies the name of the output sink topic, so I think it's fine that our automatic `-consistency` suffix doesn't mention "sink".